sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -611,6 +611,41 @@ object RapidsConf {
       .booleanConf
       .createWithDefault(false)
 
+  val ENABLE_INNER_JOIN = conf("spark.rapids.sql.join.inner.enabled")
```

Comment:
consider creating a joinType-based config generation using enumeration  https://stackoverflow.com/a/34550542/15132525 and a joinType-parametrized conf getter to avoid repetitive code throughout this PR.

Reply:
`JoinType` is a sealed abstract class that has a type hierarchy to it that does not fit with the enum pattern you listed here.

`CrossJoin` is an `InnerLikeJoin` which is a `JoinType`.

There are also JoinType sub classes that we don't deal with yet (ExistanceJoin), and ones that we will never deal with because they are just used for planning (NaturalJoin and UsingJoin).

I like the idea but I don't know how to really implement it in practice.

Reply:
Then we could list explicitly the  join types we care about:
```scala
  val joinTypeEnabledMap: Map[String, ConfEntry[Boolean]] =
    Seq(
      Inner,
      Cross,
      LeftOuter,
      ...
    ).map { jt =>
    val key = s"spark.rapids.sql.join.${jt}.enabled"
    key -> conf(key)
      .doc(s"When set to true ${jt.sql} joins are enabled on the GPU")
      .booleanConf
      .createWithDefault(true)
  }.toMap
```

then in https://github.com/NVIDIA/spark-rapids/blob/575f645690a17065ee2c5175a54fdecc2e1631a2/sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala#L1383

```scala
  def isJoinTypeEnabled(joinType: JoinType): Boolean = joinTypeEnabledMap
    .get(s"spark.rapids.sql.join.${joinType}.enabled")
    .map(confEntry => get(confEntry))
    .getOrElse(true)
```


and in https://github.com/NVIDIA/spark-rapids/pull/2285/files#diff-753a0d906f1f2d01d8bb08ef13e5205b1710cf915bb0f5fcdd55c3b665d2900dR28

```scala
object JoinTypeChecks {
  def tagForGpu(joinType: JoinType, meta: RapidsMeta[_, _, _]): Unit = {
    val conf = meta.conf
   if !conf.isJoinTypeEnabled(joinType) {
     meta.willNotWorkOnGpu("${joinType.sql} joins have been disabled. To enable set " +
            s"spark.rapids.sql.join.${joinType}.enabled" to true")
   }
}
```